rvps/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#![allow(clippy::new_without_default)]
+
+pub mod cache;
+pub mod extractors;
+pub mod pre_processor;
+pub mod reference_value;
+
+use anyhow::{anyhow, Result};
+use extractors::{Extractors, ExtractorsAPI};
+use pre_processor::{PreProcessor, PreProcessorAPI, Ware};
+use serde::{Deserialize, Serialize};
+
+pub use cache::Cache;
+pub use reference_value::ReferenceValue;
+
+/// Default version of Message
+static MESSAGE_VERSION: &str = "0.1";
+
+/// Message is an overall packet that Reference Value Provider Service
+/// receives. It will contain payload (content of different provenance,
+/// JSON format), provenance type (indicates the type of the payload)
+/// and a version number (use to distinguish different version of
+/// message, for extendability).
+/// * `version`: version of this message.
+/// * `payload`: content of the provenance, JSON encoded.
+/// * `typ`: provenance type of the payload.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Message {
+    #[serde(default = "default_version")]
+    version: String,
+    payload: String,
+    #[serde(rename = "type")]
+    typ: String,
```

Comment:
If you changed the field to `r#type`, I think you can remove the serde annotation.

Reply:
Yea, both are OK. I get accustomed to extra serde annotation because I somehow do not like `r#` here  :P.  I'll change it as your suggestion